### ISSUE TITLE
Not working with Chome

### ISSUE THREAD

**Issue by zdjeffri** [jdjaxon/linux_cac#5]
Certs are available in Chrome (visible in Security > Certs > Authorities > US Govt) however at  a CAC login, the CAC module does not seem available. Does not prompt to choose a cert or enter pin. Firefox is working fine. 

Using automatic (wget) script.

OS: Ubuntu 20.04.4 LTS
Chrome Version: Version 101.0.4951.54 (Official Build) (64-bit)

**Comment by bengooch7** [comment]
Also, run 
```
service pcscd status
```
Check that the pcscd.service and pcscd.socket are active.

**Comment by zdjeffri** [comment]
> Can you execute the following commands and post the results here?
> 
> ```shell
> ls ~/.pki/nssdb
> ```
> 
> ```shell
> cat ~/.pki/nssdb/pkcs11.txt
> ```
![image](https://user-images.githubusercontent.com/46287202/167944231-3d36b32e-a264-4fa0-b684-99baac8c8900.png)



> Also, run
> 
> ```
> service pcscd status
> ```
> 
> Check that the pcscd.service and pcscd.socket are active.

Services were up and working.

UPDATE: Problem solved by changing ownership of pkcs11.txt to user:user (from root:root).

**Comment by jdjaxon** [comment]
I'm glad you were able to figure it out. That's the first time I've seen that. You mind me asking how you installed chrome?

**Comment by zdjeffri** [comment]
> 

Nothing fancy. Downloaded the 64-bit .deb file downloaded from [https://www.google.com/chrome/downloads/](https://www.google.com/chrome/downloads/). And installed with GUI software install (right click on file in Downloads and select 'open with software install'.
![image](https://user-images.githubusercontent.com/46287202/167977118-d670dc68-66fa-46b4-b360-b9e3a8e8638f.png)

**Comment by bengooch7** [comment]
Thanks for that info. We haven't run up against this issue in any of our
testing,but I think I may have been testing by installing chrome via apt,
not the .deb package.

I will look into this today and see if I can replicate the issue.

Ben Gooch

**Comment by bengooch7** [comment]
I tested in a VM today. 
Ubuntu 20.04.1
Chrome - Version 101.0.4951.64 (Official Build) (64-bit)

As you stated, I installed by downloading the .deb from Google's site (using Firefox) and then right-clicked the .deb, and selected "Open with Software Install". 
I then ran Chrome once to generate the pkcs11 file.
The owner set on the file was user:user.

The only way I was able to replicate the root:root owner was to run 
```
sudo google-chrome --no-sandbox
```
and even then, the owner/permissions were only applicable to the 
```
/root/.pki/nssdb/pkcs11
```
version of the file.

I've also tried installing chrome via apt (`sudo apt install google-chrome`) which still yields a user:user owner.

I'm not sure how you happened across having the pkcs11 file owned by root, unless it was set like that by some other program you have that uses the same files in the .pki folder of the user's home directory.

Thanks for the info you've provided, and if you figure out how it happened, I'd love to know so I can ensure the script corrects that issue.

-Ben
